### ISSUE TITLE
Redhat(Openshift) support

### ISSUE THREAD

**Issue by andyzhangx** [kubernetes-sigs/azuredisk-csi-driver#658]
**Is your feature request related to a problem?/Why is this needed**

**Describe the solution you'd like in detail**

To support Redhat, ssl mount path should be `/etc/pki/tls/certs`, need to make this ssl mount path configurable in charts:
https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/d95c8ebc51834ec63493495c4c359ba5a6b98a00/charts/latest/azuredisk-csi-driver/templates/csi-azuredisk-node.yaml#L138-L140

 - also need workaround in Redhat by copying `ca-certificate.crt`:

```console
copy /etc/pki/tls/certs/ca-bundle.crt /etc/pki/tls/certs/ca-certificate.crt
```

original cert related error in Redhat:
```
I0105 08:01:19.630448       1 azure_retry.go:182] Backoff retrying PUT "https://management.azure.com/subscriptions/xxx/resourceGroups/woc-azure-lab-91/providers/Microsoft.Compute/disks/pvc-5bb9d91f-f220-4d57-bf1a-21329a237fee?api-version=2019-11-01" with error &{true -1 0001-01-01 00:00:00 +0000 UTC azure.BearerAuthorizer#WithAuthorization: Failed to refresh the Token for request to https://management.azure.com/subscriptions/xxx/resourceGroups/woc-azure-lab-91/providers/Microsoft.Compute/disks/pvc-5bb9d91f-f220-4d57-bf1a-21329a237fee?api-version=2019-11-01: StatusCode=0 -- Original Error: adal: Failed to execute the refresh request. Error = 'Post "https://login.microsoftonline.com/93f5baf9-414a-4f1b-88bc-33f3013923d7/oauth2/token": x509: certificate signed by unknown authority'}
```

**Describe alternatives you've considered**

**Additional context**

refer to https://github.com/Azure/OpenShift/issues/216

**Comment by andyzhangx** [comment]
also related to https://github.com/kubernetes-sigs/azuredisk-csi-driver/issues/610

**Comment by ezYakaEagle442** [comment]
@andyzhangx as it looks like the Driver still look for the CA certs. at/etc/ssl/certs , in fact the issue there is that the Intermediate CA "DigiCert SHA2 Secure Server CA" is missing at at/etc/ssl/certs : the X509 error is about the certificate "login.microsoftonline-int.com" delivered on 06 Oct. 2020 by "DigiCert SHA2 Secure Server CA" which parent is "DigiCert Global Root CA"

ARO Test from a pod allowed to access HostPath as I explained in #662  : 
```sh
ls -al /etc/ssl/certs | grep -i "DigiCert"
```

```console
lrwxrwxrwx. 1 root root     38 Dec 15 20:20 244b5494.0 -> DigiCert_High_Assurance_EV_Root_CA.pem
lrwxrwxrwx. 1 root root     27 Dec 15 20:20 3513523f.0 -> DigiCert_Global_Root_CA.pem
lrwxrwxrwx. 1 root root     27 Dec 15 20:20 607986c7.0 -> DigiCert_Global_Root_G2.pem
lrwxrwxrwx. 1 root root     28 Dec 15 20:20 75d1b2ed.0 -> DigiCert_Trusted_Root_G4.pem
lrwxrwxrwx. 1 root root     31 Dec 15 20:20 7f3d5d1d.0 -> DigiCert_Assured_ID_Root_G3.pem
lrwxrwxrwx. 1 root root     31 Dec 15 20:20 9d04f354.0 -> DigiCert_Assured_ID_Root_G2.pem
lrwxrwxrwx. 1 root root     66 Dec 15 20:20 DigiCert_Assured_ID_Root_CA.pem -> /usr/share/ca-certificates/mozilla/DigiCert_Assured_ID_Root_CA.crt
lrwxrwxrwx. 1 root root     66 Dec 15 20:20 DigiCert_Assured_ID_Root_G2.pem -> /usr/share/ca-certificates/mozilla/DigiCert_Assured_ID_Root_G2.crt
lrwxrwxrwx. 1 root root     66 Dec 15 20:20 DigiCert_Assured_ID_Root_G3.pem -> /usr/share/ca-certificates/mozilla/DigiCert_Assured_ID_Root_G3.crt
lrwxrwxrwx. 1 root root     62 Dec 15 20:20 DigiCert_Global_Root_CA.pem -> /usr/share/ca-certificates/mozilla/DigiCert_Global_Root_CA.crt
lrwxrwxrwx. 1 root root     62 Dec 15 20:20 DigiCert_Global_Root_G2.pem -> /usr/share/ca-certificates/mozilla/DigiCert_Global_Root_G2.crt
lrwxrwxrwx. 1 root root     62 Dec 15 20:20 DigiCert_Global_Root_G3.pem -> /usr/share/ca-certificates/mozilla/DigiCert_Global_Root_G3.crt
lrwxrwxrwx. 1 root root     73 Dec 15 20:20 DigiCert_High_Assurance_EV_Root_CA.pem -> /usr/share/ca-certificates/mozilla/DigiCert_High_Assurance_EV_Root_CA.crt
lrwxrwxrwx. 1 root root     63 Dec 15 20:20 DigiCert_Trusted_Root_G4.pem -> /usr/share/ca-certificates/mozilla/DigiCert_Trusted_Root_G4.crt
lrwxrwxrwx. 1 root root     31 Dec 15 20:20 b1159c4c.0 -> DigiCert_Assured_ID_Root_CA.pem
lrwxrwxrwx. 1 root root     27 Dec 15 20:20 dd8e9d41.0 -> DigiCert_Global_Root_G3.pem
```

RedHat ships the [Mozilla certificates](https://ccadb-public.secure.force.com/mozilla/IncludedCACertificateReport), and you can check the "DigiCert SHA2 Secure Server CA" is not on their official list ....

**Comment by ezYakaEagle442** [comment]
It is not shipped neither in Bundle you suggested to copy as a workaround : 

```sh
cat /mnt/pki/tls/certs/ca-certificate.crt | grep -i "DigiCert"
```

```console
root@pki-tls-ca-cnf-pod:/# cat /mnt/pki/tls/certs/ca-certificate.crt | grep -i "DigiCert"
# DigiCert Assured ID Root CA
# DigiCert Assured ID Root G2
# DigiCert Assured ID Root G3
# DigiCert Global Root CA
# DigiCert Global Root G2
# DigiCert Global Root G3
# DigiCert High Assurance EV Root CA
# DigiCert Trusted Root G4
```

**Comment by ezYakaEagle442** [comment]
interesting test:

```sh
curl --cacert non_existing_file https://login.microsoftonline.com
```

```console
curl: (77) error setting certificate verify locations:
  CAfile: non_existing_file
  CApath: /etc/ssl/certs
```

The above confirms curl uses /etc/ssl/certs location.

```sh
curl https://login.microsoftonline.com
```

```console
<html><head><title>Object moved</title></head><body>
<h2>Object moved to <a href="https://www.office.com/login#">here</a>.</h2>
</body></html>
```

Strangely curl that uses the same CA location as the CSI Driver but does validate the certificate without any error.

**Comment by ezYakaEagle442** [comment]
I have also found [https://github.com/coreos/bugs/issues/2262](https://github.com/coreos/bugs/issues/2262)

the Go [x509 modules hardcodes](https://golang.org/src/crypto/x509/root_linux.go) /etc/ssl/certs/ca-certificates.crt

**Comment by ezYakaEagle442** [comment]
@andyzhangx the final intesresting test which reflect our Driver use case & behaviour : 

```sh
curl --capath /mnt/pki/tls/certs/ca-certificate.crt  https://login.microsoftonline.com
```

```console
curl: (60) SSL certificate problem: unable to get local issuer certificate
More details here: https://curl.haxx.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.
```

**Comment by ezYakaEagle442** [comment]
I have also found this outdated [RedHat blog](https://www.openshift.com/blog/considerations-on-openshift-pkis-and-certificates) refering to /etc/origin which is now empty on worker nodes.  Also this outdated [OCP doc](https://access.redhat.com/documentation/en-us/openshift_container_platform/4.3/html/authentication/ocp-certificates)

My point is just that the CA file location should be configurable from the CSI Driver  to manage any change from RedHat / ARO / CoreOS Linux side.

**Comment by andyzhangx** [comment]
This issue is introduced by https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/600, which is only for Azure stack, remove following mount should work:

```yaml
name: ssl
  mountPath: /etc/ssl/certs
  readOnly: true
```

**Comment by andyzhangx** [comment]
I have also pushed the fix to release `v0.10.0` and master branch, so this issue should be fixed now. And for the `azure-cloud-provider`  secret setting in ARO, I think we need to add a notion in this project.

**Comment by andyzhangx** [comment]
there should be doc update for ARO, refer to https://github.com/ezYakaEagle442/aro-pub-storage/blob/master/setup-store-CSI-driver-azure-disk.md
